### PR TITLE
Unit assignment for referrals

### PR DESCRIPTION
**Pull request by gigxz**
https://www.pivotaltracker.com/n/projects/2591838/stories/185510437

also return API-specific error message to the frontend so it can display whether the error is related to a connection to LINK/MCI vs an internal error. hoping these don't come up, but in case we have issues with timeouts or unreliability.

### CODE REVIEW

**Comment by gigxz** [comment]
This has been deployed and tested with link

File: drivers/hmis/app/graphql/types/forms/pick_list_option.rb [hunk 1]
```diff
@@ -151,7 +151,13 @@ def self.options_for_type(pick_list_type, user:, relation_id: nil)
       end
     end
 
+    def self.all_unit_types
+      Hmis::UnitType.order(:description, :id).map(&:to_pick_list_option)
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/forms/pick_list_option.rb:155]: Is there, or should there be an "active" or "inactive" flag/scope on UnitType?

**Review by eanders — APPROVED** [review]
Looks great!

File: drivers/hmis/app/graphql/types/forms/pick_list_option.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/forms/pick_list_option.rb:155]: There is not. The Hmis::ProjectUnitTypeMapping maps Projects=>Allowed Unit Types and that table does have a active/inactive flag, but the Unit Types themselves don't. I think it's ok for now we do not have a use case for it.

**Pull request merged**